### ISSUE TITLE
Run wav2vec on larger audios

### ISSUE THREAD

**Issue by kruthikakr** [pytorch/fairseq#3462]
Hi 
I am trying to run 
Use wav2vec 2.0 with hugsTransformers:

#### Code
# !pip install transformers
import soundfile as sf
import torch
from transformers import Wav2Vec2ForMaskedLM, Wav2Vec2Tokenizer

# load pretrained model
tokenizer = Wav2Vec2Tokenizer.from_pretrained("facebook/wav2vec2-base-960h")
model = Wav2Vec2ForMaskedLM.from_pretrained("facebook/wav2vec2-base-960h")

# load audio
audio_input, _ = sf.read("path/to/audio/file")

# transcribe
input_values = tokenizer(audio_input, return_tensors="pt").input_values
logits = model(input_values).logits
predicted_ids = torch.argmax(logits, dim=-1)
transcription = tokenizer.batch_decode(predicted_ids)[0]

#### What have you tried?
What should be the duration of the audio file ? if the audio is bigger say > 2min wav file ? how to run the model ?

**Comment by harveenchadha** [comment]
To run the model on large files  make sure you break the larger audios into smaller audios using Voice Activity Detection (VAD)  as described [here](https://github.com/wiseman/py-webrtcvad) and run the model on smaller chunks and finally concatenate the output at the end.

**Comment by kruthikakr** [comment]
Can i use the same VAD for different languages ?

**Comment by harveenchadha** [comment]
Yes VAD is not dependent on Language, it works on signal level.

**Comment by kruthikakr** [comment]
Thanks I will check out
